torch/backends/cudnn/__init__.py

```diff
@@ -1,91 +1,51 @@
-    # 2. Default directories
-    #    That is stored in the environment variable `PATH`.
-    test_env = os.environ.copy()
-    old_path = test_env['PATH']
-    th_dll_path = os.path.join(os.path.dirname(
-        os.path.dirname(os.path.dirname(__file__))), 'lib')
-    test_env['PATH'] = ';'.join([th_dll_path, old_path])
-    proc = Popen(['where', 'cudnn64*.dll'], stdout=PIPE,
-                 stderr=PIPE, stdin=PIPE, env=test_env)
-    out, err = proc.communicate()
-    out = out.decode().strip()
-    if len(out) > 0:
-        if out.find('\r\n') != -1:
-            out = out.split('\r\n')[0]
-        cudnn_lib = str(out)
-        return ctypes.cdll.LoadLibrary(cudnn_lib)
-    else:
-        return None
 
-
-def _libcudnn():
-    global lib, __cudnn_version
-    if lib is None:
-        if sys.platform == "win32":
-            lib = find_cudnn_windows_lib()
-        else:
-            lib = ctypes.cdll.LoadLibrary(None)
-        if hasattr(lib, 'cudnnGetErrorString'):
-            lib.cudnnGetErrorString.restype = ctypes.c_char_p
-            __cudnn_version = lib.cudnnGetVersion()
-            compile_version = torch._C._cudnn_version()
-            # cuDNN version is MAJOR*1000 + MINOR*100 + PATCH
-            runtime_major = __cudnn_version // 1000
-            runtime_minor = (__cudnn_version % 1000) // 100
-            compile_major = compile_version // 1000
-            compile_minor = (compile_version % 1000) // 100
+if _cudnn is not None:
+    def _init():
+        global __cudnn_version
+        if __cudnn_version is None:
+            __cudnn_version = _cudnn.getVersionInt()
+            runtime_version = _cudnn.getRuntimeVersion()
+            compile_version = _cudnn.getCompileVersion()
+            runtime_major, runtime_minor, _ = runtime_version
+            compile_major, compile_minor, _ = compile_version
             # Different major versions are always incompatible
             # Starting with cuDNN 7, minor versions are backwards-compatible
             if runtime_major != compile_major:
                 cudnn_compatible = False
-            elif runtime_major < 7:
+            elif runtime_major < 7 or not _cudnn.is_cuda:
                 cudnn_compatible = runtime_minor == compile_minor
             else:
                 cudnn_compatible = runtime_minor >= compile_minor
             if not cudnn_compatible:
                 raise RuntimeError(
                     'cuDNN version incompatibility: PyTorch was compiled against {} '
-                    'but linked against {}'.format(compile_version, __cudnn_version))
-        elif hasattr(lib, 'miopenGetVersion'):
-            miopen_major = ctypes.c_size_t()
-            miopen_minor = ctypes.c_size_t()
-            miopen_patch = ctypes.c_size_t()
-            # miopen version is MAJOR*1000000 + MINOR*1000 + PATCH
-            lib.miopenGetVersion(ctypes.byref(miopen_major), ctypes.byref(miopen_minor), ctypes.byref(miopen_patch))
```

Comment:
@jithunnair-amd I see you added this in #33118 but the call doesn't seem to compile when I move it into C++.
```
17:55:36 /var/lib/jenkins/workspace/torch/csrc/cuda/shared/cudnn.cpp:43:3: error: use of undeclared identifier 'miopenGetVersion'
17:55:36   miopenGetVersion(&major, &minor, &patch);
```

Reply:
Yes, that's because the `miopenGetVersion` api will be part of the next release of MIOpen. However, in the python context, we use a guard `elif hasattr(lib, 'miopenGetVersion'):` to make sure the api is present.

Reply:
Okay, so it will be available in version 2.3.0?

Reply:
Yes.